packages/diagrams/frontend/sirius-components-diagrams/src/sprotty/views/InsideLabelView.tsx

```diff
@@ -0,0 +1,107 @@
+import { setAttr, setNamespace, SLabelView, svg } from 'sprotty';
+import { getSubType } from 'sprotty-protocol';
+const preventRemovalOfUnusedImportByPrettier = svg !== null;
+
+/**
+ * The view used to display inside labels.
+ *
+ * @arichard
+ */
+export class InsideLabelView extends SLabelView {
+  // @ts-ignore
+  render(label) {
+    const { color, bold, underline, strikeThrough, italic, fontSize, iconURL, opacity } = label.style;
+
+    const width: number = label.parent.size.width - label.position.y * 2;
+    const height: number = label.parent.size.height - label.position.x * 2;
+
+    // The font-family is hardcoded to match with the backend compute bounds algo.
+    const textStyle = {
+      width: width + 'px',
+      height: height + 'px',
+      color: color,
+      opacity: opacity,
+      'font-size': fontSize + 'px',
+      'font-family': 'Arial, Helvetica, sans-serif',
+      'font-weight': 'normal',
+      'font-style': 'normal',
+      'overflow-wrap': 'anywhere',
+      'white-space': 'break-spaces',
+      display: 'flex',
+      'flex-wrap': 'no-wrap',
+    };
+    if (bold) {
+      textStyle['font-weight'] = 'bold';
+    }
+    if (italic) {
+      textStyle['font-style'] = 'italic';
+    }
+    if (underline) {
+      textStyle['text-decoration'] = 'underline';
+    }
+    if (strikeThrough) {
+      if (textStyle['text-decoration'] === 'none') {
+        textStyle['text-decoration'] = 'line-through';
+      } else {
+        textStyle['text-decoration'] += ' line-through';
+      }
+    }
+    if (label.type.endsWith('-h_left')) {
+      textStyle['text-align'] = 'left';
+      textStyle['justify-content'] = 'flex-start';
+    } else if (label.type.endsWith('-h_right')) {
+      textStyle['text-align'] = 'right';
+      textStyle['justify-content'] = 'flex-end';
+    } else {
+      textStyle['text-align'] = 'center';
+      textStyle['justify-content'] = 'center';
+    }
+    if (label.type.includes('-v_top')) {
+      textStyle['align-items'] = 'flex-start';
+    } else if (label.type.includes('-v_bottom')) {
+      textStyle['align-items'] = 'flex-end';
+    } else {
+      textStyle['align-items'] = 'center';
+    }
+
+    const foreignObjectContents: JSX.Element = (
+      <div style={textStyle}>
+        {iconURL ? <img height="16" width="16" alt="" src={iconURL} /> : ''}
+        {label.text}
+      </div>
+    );
+    const vnode: JSX.Element = (
+      <foreignObject
+        requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"
+        height={height}
+        width={width}
+        class-sprotty-label={true}
+        attrs-data-testid={`Label - ${label.text}`}
+        style={{ pointerEvents: 'none' }}>
+        {foreignObjectContents}
+      </foreignObject>
+    );
+    const subType: string = getSubType(label);
+    if (subType) {
+      // @ts-ignore
+      setAttr(vnode, 'class', subType);
+      // @ts-ignore
+      setNamespace(foreignObjectContents, 'http://www.w3.org/1999/xhtml');
```

Comment:
Why `@ts-ignore`? Is that related to the fact that the type of the variables is wrong `VNode` vs `JSX.Element`?

Reply:
I'll open an issue to try to improve our typing in the Sprotty parts of the code, you can keep the code as is for now. It's an issue found in most of our Sprotty views.